DSharpPlus/Entities/Interaction/Application/DiscordAutoCompleteChoice.cs

```diff
@@ -0,0 +1,61 @@
+// This file is part of the DSharpPlus project.
+//
+// Copyright (c) 2015 Mike Santiago
+// Copyright (c) 2016-2021 DSharpPlus Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+using System;
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities
+{
+    /// <summary>
+    /// Represents an option for a user to select for auto-completion.
+    /// </summary>
+    public sealed class DiscordAutoCompleteChoice
+    {
+        /// <summary>
+        /// Gets the name of this option which will be presented to the user.
+        /// </summary>
+        [JsonProperty("name")]
+        public string Name { get; internal set; }
+
+        /// <summary>
+        /// Gets the value of this option.
+        /// </summary>
+        [JsonProperty("value")]
+        public string Value { get; internal set; }
```

Comment:
Autcomplete works for both string and integer option types, so this should be `object` and check for `string` and `long` in the constructor.

Reply:
I was aware of this prior to opening the PR (as it's been question on ddevs), but wasn't sure if I should go that route. I suppose JSON.NET could potentially break with the exception of `Cannot convert System.Int32 into System.String` or the likes.

Reply:
JSON.NET will actually swallow this fine - it's the only serializer to do so.

Reply:
I'll be fixing this momentarily.